src/main/java/net/mcreator/ui/blockly/BlocklyJavascriptBridge.java

```diff
@@ -239,6 +241,30 @@ public void addExternalTrigger(ExternalTrigger external_trigger) {
 		ext_triggers.put(external_trigger.getID(), external_trigger.getName());
 	}
 
+	@SuppressWarnings("unused") public boolean hasDependency(String procedureName, String dependencyName) {
+		ModElement me = mcreator.getWorkspace().getModElementByName(procedureName);
+		return me != null && me.getGeneratableElement() instanceof Procedure procedure && procedure.getDependencies()
+				.stream().anyMatch(e -> e.getName().equals(dependencyName));
+	}
+
+	@SuppressWarnings("unused") public String getDependencyType(String procedureName, String dependencyName) {
+		ModElement me = mcreator.getWorkspace().getModElementByName(procedureName);
+		if (me != null && me.getGeneratableElement() instanceof Procedure procedure) {
+			Optional<Dependency> dependency = procedure.getDependencies().stream()
+					.filter(e -> e.getName().equals(dependencyName)).findFirst();
+			if (dependency.isPresent() && VariableTypeLoader.INSTANCE.fromName(dependency.get().getRawType()) != null)

```

Comment:
Looking back into the PR, I'm wondering if it would be better to return a "dummy" Blockly type for dependencies that don't have a variable type. This would allow connections only with `null` blocks (or blocks of the dummy type), without needing to disable the procedure block

Reply:
Like, to be able to pass e.g. null `world` or `guistate` to a procedure?

Reply:
By `null` blocks I mean blocks like the custom code snippet, which has a `null` output type

Reply:
Good idea, but not sure how such dummy type can be implemented. Maybe just keep the block valid if dependency has no known type but input block does not as well (like code snippets you mentioned)?

Reply:
I guess we could return something like `"Unknown"` if the variable type from name is null (under the assumption that people don't add procedures with output type `"Unknown"`)

Reply:
I would not assume that, this is just a regular string and also a bit generic name.
I also looked into the code and tested it again again and I think one already can do what you suggested. Adding an action result type to `world` dependency made the block invalid, whereas code snippet did not. However, I can adapt it to handle `'Null'` type output blocks the same way.

Reply:
The point of this is that we wouldn't need to make the procedure block invalid at all, making the JS code simpler. It's also more intuitive to not be able to plug e.g. action result to `world` at all

Reply:
Hmm, in this case I think I'm aware of an even more elegant solution without the need for a placeholder type name.

Reply:
✔️ Fixed